src/test_report.py

```diff
@@ -98,11 +101,11 @@ def _get_group_data(self, root_node):
 
     def _get_results_data(self, root_node):
         group_nodes = self._db.get_nodes({"parent": root_node['_id']})
```

Comment:
I think we can also avoid doing `get_nodes()` here.  In fact, the only nodes we need to retrieve from the database are the top-level one to get the revision details and the test cases that failed.

Reply:
We need to use `get_nodes()` here to get all the child nodes of `checkout`. Need to use `count_nodes()` in `_get_group_data()` method instead. Update the code accordingly.

Reply:
Tested on local.

```
kernelci-pipeline-test_report | mainline/master v6.0-rc6: 3 runs 2 failures
kernelci-pipeline-test_report | 
kernelci-pipeline-test_report | Summary
kernelci-pipeline-test_report | =======
kernelci-pipeline-test_report | 
kernelci-pipeline-test_report | Tree:     mainline
kernelci-pipeline-test_report | Branch:   master
kernelci-pipeline-test_report | Describe: v6.0-rc6
kernelci-pipeline-test_report | URL:      https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git
kernelci-pipeline-test_report | SHA1:     521a547ced6477c54b4b0cc206000406c221b4d6
kernelci-pipeline-test_report | 
kernelci-pipeline-test_report | Name            | Result   | Total    | Failures
kernelci-pipeline-test_report | ----------------+----------+----------+---------
kernelci-pipeline-test_report | kver            | pass     |        0 |        0
kernelci-pipeline-test_report | kunit           | fail     |        1 |        1
kernelci-pipeline-test_report | kunit-test-case | fail     |        1 |        1
kernelci-pipeline-test_report | 
kernelci-pipeline-test_report | 
kernelci-pipeline-test_report | Failing tests
kernelci-pipeline-test_report | =============
kernelci-pipeline-test_report | 
kernelci-pipeline-test_report | kunit
kernelci-pipeline-test_report | -----
kernelci-pipeline-test_report | 
kernelci-pipeline-test_report | * kunit-test-case
kernelci-pipeline-test_report | 
kernelci-pipeline-test_report | 
kernelci-pipeline-test_report | kunit-test-case
kernelci-pipeline-test_report | ---------------
kernelci-pipeline-test_report | 
kernelci-pipeline-test_report | * 
```